ctapipe/core/qualityquery.py

```diff
@@ -25,38 +33,45 @@ class QualityQuery(Component):
     """
 
     quality_criteria = List(
-        Tuple(Unicode(), Unicode()),
-        help=(
-            "list of tuples of ('<description', 'expression string') to accept "
-            "(select) a given data value.  E.g. ``[('mycut', 'x > 3'),]``. "
-            "You may use ``numpy`` as ``np`` and ``astropy.units`` as ``u``,"
-            " but no other modules."
-        ),
+        Tuple(Unicode(), TelescopeParameter(Unicode())),
+        help="List of tuples of ('query name', TelescopeParameter or expression) to accept"
+        " (select) a given data value. Example: ``[('positive', 'x > 0')]`` or"
+        " ``[('high_intensity', [('type', '*', 'hillas_intensity > 50'), ('type', 'SST*', 'hillas_intensity > 100')])]``"
+        " You may use ``numpy`` as ``np`` and ``astropy.units`` as ``u``,"
+        " but no other modules.",
     ).tag(config=True)
 
-    def __init__(self, config=None, parent=None, **kwargs):
-        super().__init__(config=config, parent=parent, **kwargs)
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+
+        self._compile()
+
+    def _compile(self):
+        self.engines = []
+        self.criteria_names = []
+
+        for name, criteria_list in self.quality_criteria:
+
+            if not isinstance(criteria_list, (tuple, list, TelescopePatternList)):
```

Comment:
How can this happen? The traitlets system should make sure that this is always a `TelescopePatternList`

Reply:
Yes, it should, but I can't seem to get it to work correctly.

Reply:
fixed